app/assets/javascripts/discourse/tests/integration/components/themes-list-test.js

```diff
@@ -179,18 +184,90 @@ module("Integration | Component | themes-list", function (hooks) {
       hbs`<ThemesList @themes={{this.themes}} @components={{(array)}} @currentTab={{this.currentTab}} />`
     );
 
-    assert.ok(exists(".themes-list-filter"));
-    await fillIn(".themes-list-filter .filter-input", "  oSAma ");
+    assert.ok(exists(".themes-list-container__search-input"));
+    await fillIn(".themes-list-container__search-input", "  oSAma ");
     assert.deepEqual(
-      Array.from(queryAll(".themes-list-item .name")).map((node) =>
-        node.textContent.trim()
+      Array.from(queryAll(".themes-list-container__item .info .name")).map(
+        (node) => node.textContent.trim()
       ),
       ["Theme osama", "Theme OsAmaa", "Theme osAMA 1234"],
-      "only themes whose names include the filter term are shown"
+      "only themes whose names include the search term are shown"
+    );
+  });
+
+  test("themes filter", async function (assert) {
+    const themes = [
+      Theme.create({ name: "Theme enabled 1", user_selectable: true }),
+      Theme.create({ name: "Theme enabled 2", user_selectable: true }),
+      Theme.create({ name: "Theme disabled 1", user_selectable: false }),
+      Theme.create({
+        name: "Theme disabled 2",
+        user_selectable: false,
+        remote_theme: {
+          id: 42,
+          remote_url:
+            "git@github.com:discourse-org/discourse-incomplete-theme.git",
+          commits_behind: 1,
+        },
+      }),
+    ];
+    this.setProperties({
+      themes,
+      currentTab: THEMES,
+    });
+
+    await render(
+      hbs`<ThemesList @themes={{this.themes}} @components={{(array)}} @currentTab={{this.currentTab}} />`
+    );
+
+    assert.ok(exists(".themes-list-container__filter-input"));
+    assert.deepEqual(
+      Array.from(queryAll(".themes-list-container__item .info .name")).map(
+        (node) => node.textContent.trim()
+      ),
+      [
+        "Theme enabled 1",
+        "Theme enabled 2",
+        "Theme disabled 1",
+        "Theme disabled 2",
+      ]
+    );
+
+    await selectKit(".themes-list-container__filter-input").expand();
+    await selectKit(".themes-list-container__filter-input").selectRowByValue(
+      "active"
+    );
+    assert.deepEqual(
+      Array.from(queryAll(".themes-list-container__item .info .name")).map(
```

Comment:
Ah I thought there might be a slightly nicer way than doing `Array.from`, and there is:

```javascript
[...queryAll(".themes-list-container__item .info .name")].map((el) => el.innerText.trim()),
```